Source/Controllers/DefaultAsyncMessagesCollectionViewDataSource.swift

```diff
@@ -37,14 +38,14 @@ class DefaultAsyncMessagesCollectionViewDataSource: NSObject, AsyncMessagesColle
     }
 
     //MARK: ASCollectionViewDataSource methods
-    func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+    func collectionNode(_ collectionNode: ASCollectionNode, numberOfItemsInSection section: Int) -> Int {
         assert(nodeMetadatas.count == messages.count, "Node metadata is required for each message.")
         return messages.count
     }
 
     //TODO Use node block
-    func collectionView(_ collectionView: ASCollectionView, nodeForItemAt indexPath: IndexPath) -> ASCellNode {
```

Comment:
Not a blocking issue, but let's aim to switch to the node block method (`collectionNode(collectionNode:nodeBlockForItem:`) soon. It allows nodes to be allocated concurrently which is much faster, especially for a large data set.

Reply:
Great! That would be much better